### ISSUE TITLE
Failed to reproduce results on MS COCO val set

### ISSUE THREAD

**Issue by sry2** [zhang-haojie/wesam#18]
Appreciate for the work and codes again. When I tried to reproduce the paper's result on COCO following the instruction of README and descriptions in the paper, I found the adapted model hard to reach ```80.12``` mIoU reported in the paper. The results of my multiple experiments (repeated for several times, on COCO val, following the code's settings) varied around 76.5 but never reached 77 or better performance. I checked the configs of training to make sure the learning rate and epochs are correct (1e-4, 5). Also, I tried through total bsz 1, 4 and 8 (with lr scaled with it) but still failed to improve the results. Is there something might be missed by my experiments? 

I also have a question on ```opt.steps``` settings in the config. If ```epoch=5```, for COCO val set the step scheduler will be meaningless because the training will never reach  5000 steps. Does this setting require modification regarding different datasets?

My settings for 4 GPUs (total bsz: 4):
```python3
config = {
    "gpu_ids": "0,1,2,3",
    "batch_size": 1,
    "val_batchsize": 4,
    "num_workers": 4,
    "num_epochs": 5,
    "max_nums": 50,
    "num_points": 5,
    "resume": False,
    "dataset": "COCO",
    "visual": False,
    "load_type": "soft",
    "prompt": "box",
    "out_dir": "output/COCO/wesam/",
    "name": "base",
    "corrupt": None,
    "visual": False,
    "opt": {
        "learning_rate": 1e-4,
    }
}
```

**Comment by xixinzhang** [comment]
I've also tried to reproduce the result on COCO and got similar results, with mIoU around 76.4.

**Comment by GZ-YourZY** [comment]
May I ask how your inference code is configured?

**Comment by sry2** [comment]
> May I ask how your inference code is configured?
I didn't set extra configs on inference code but just used the official codes provided in this repo. The training codes ```adaptation.py``` contains periodic inference and evaluation every epoch.

**Comment by zhang-haojie** [comment]
In the recent commit, I fixed an error. After the update, most of the other datasets have been improved, except for the reduction of the indicators of the COCO dataset. The IoU tested by the COCO dataset is also close to `77`, compared with the previous `79~80`.

We suspect that the COCO dataset is not very accurately annotated ([SAM](http://arxiv.org/abs/2304.02643) also reveals that some dataset annotations are not precise). The other datasets we used possess higher precision annotations, visually. Here, we recommend the [COCONut](https://github.com/bytedance/coconut_cvpr2024) Dataset

**Comment by sry2** [comment]
Thanks for the reply. What I am expecting actually is guaranteeing the method's effectiveness. Since the experiment COCO is the mostly referred one in multiple comparisons and ablation studies in the paper, I regard it as the mostly representative experiment and look for reproducing the results. But it seems impractical now. I wonder may I ask for your recommendation on reliable dataset that can be used to solidly validate the effectiveness of the method (like, maybe some datasets in medical or some other scenes mentioned in the paper, thus there are results to be referred to)? I understand the advantage of [COCONut](https://github.com/bytedance/coconut_cvpr2024) dataset but the paper did not report the performance of any experiment on it so that is why I am expecting other available choices.

**Comment by zhang-haojie** [comment]
For natural images, you can refer to the `PascalVOC` dataset. For medical images, the `Kvasir-SEG` dataset is recommended. For camouflaged objects, the `CAMO` dataset is suitable.
